enos/modules/worker/main.tf

```diff
@@ -0,0 +1,204 @@
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+data "enos_environment" "current" {}
+
+locals {
+  selected_az = data.aws_availability_zones.available.names[random_integer.az.result]
+  common_tags = merge(
+    var.common_tags,
+    {
+      Type = var.cluster_tag,
+    },
+  )
+}
+
+resource "random_pet" "worker" {
+  separator = "_"
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+  filter {
+    name   = "zone-name"
+    values = var.availability_zones
+  }
+}
+
+data "aws_availability_zone" "worker_az" {
+  name = local.selected_az
+}
+
+data "aws_kms_key" "kms_key" {
+  key_id = var.kms_key_arn
+}
+
+resource "random_integer" "az" {
+  min = 0
+  max = length(data.aws_availability_zones.available.names) - 1
+  keepers = {
+    # Generate a new integer each time the list of aws_availability_zones changes
+    # keepers have to be strings, sort the list in case order changes but zones don't
+    listener_arn = join("", sort(data.aws_availability_zones.available.names))
+  }
+}
+
+# Create a subnet so that the worker doesn't share one with a controller
+resource "aws_subnet" "default" {
+  vpc_id                  = var.vpc_name
+  cidr_block              = "10.13.9.0/24"
+  map_public_ip_on_launch = true
+  availability_zone       = local.selected_az
+  tags = merge(
+    local.common_tags,
+    {
+      "Name" = "${var.vpc_name}_worker_subnet"
+    },
+  )
+}
+
+# The worker instance is a part of this security group, not to be confused with the next rule below
+resource "aws_security_group" "default" {
+  name        = "boundary-sg-worker-${random_pet.worker.id}"
+  description = "SSH to worker to KMS and controllers"
+  vpc_id      = var.vpc_name
+
+  ingress {
+    description = "SSH to the worker instance"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["${data.enos_environment.current.public_ip_address}/32"]
+  }
+
+  egress {
+    description = "Communication from Boundary worker to controller"
+    from_port   = 9201
+    to_port     = 9201
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    description = "Communication from Boundary worker to controller"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = merge(
+    local.common_tags,
+    {
+      "Name" = "${var.vpc_name}_worker_sg"
+    },
+  )
+}
+
+# This module only manages a rule for the controller SG that is vital to the worker's operation
+# This module does _not_ manage the security group itself
+resource "aws_vpc_security_group_ingress_rule" "worker_to_controller" {
+  description       = "This rule allows traffic from a worker to controllers over WAN"
+  security_group_id = var.controller_sg_id
+  cidr_ipv4         = "${aws_instance.worker.public_ip}/32"
+  from_port         = 9201
+  to_port           = 9201
+  ip_protocol       = "tcp"
+}
+
+data "aws_route_table" "default" {
```

Comment:
Maybe a comment here, we're fetching the existing route tables from the `infra` module, so we can add a new route for our isolated subnet